src/print.lisp

```diff
@@ -381,4 +589,8 @@
       (t
        (write-string res destination)))))
 
+(defmacro with-input-from-string ((stream string) &body body)
```

Comment:
This works (given the streams = cons implementation) but is used only in the tests at present.

If I recall correctly, one or more test cases copied from the CLHS used it. I do not at this moment recall if they were tests that would actually pass, though.